### DISCUSSION TITLE
Model rendering at low resolution

### DISCUSSION THREAD

**Discussion by tom-deadline** [google/model-viewer#2116] [Q&A]
Hey guys, you have created a great product, and thank you very much for that.
I have a question about the resolution that models are rendering at. I have a model of Marshall speaker and its grill is kind of flickering when I play with it with the camera controls ([video](https://drive.google.com/file/d/18d22luEUBxhFlb3EHmU69WPyPFbp9Kyc/view?usp=sharing)). I thought that it can be something related to the resolution. 
Could you help me with it?

**Comment by elalish** [comment]
What you're experiencing is called aliasing. We do a lot of work in the renderer to avoid/reduce it, but it's very difficult in some situations. However, sometimes this happens just because of poor glTF export. Verify that your glTF does not specify any `minFilter` or `magFilter` under `Samplers` (you can check this in a text editor, or in the last tab on our [editor](https://modelviewer.dev/editor/)). The default values are made to reduce aliasing; specifying anything will generally make it worse.

**Reply by tom-deadline** [reply]
I've tried to remove those configs from the glTF file, but the result is the same. How can I find out at what resolution the model is being rendered?

**Reply by elalish** [reply]
As long as it's not moving, it's always rendered at screen resolution. If the model is too heavy for your GPU, it'll reduce resolution to keep the frame rate up while it moves.

**Reply by tom-deadline** [reply]
is there any way to set antialiasing to true in the model viewer?

**Reply by elalish** [reply]
We do lots of antialiasing by default. Are you sure your glTF is correct?

**Reply by tom-deadline** [reply]
what do you mean by "correct"?

**Reply by tom-deadline** [reply]
Additional info: on phones the quality is much better

**Reply by elalish** [reply]
We're having occasional reports of strange resolution problems on Windows machines; is that your environment by chance? Have you tested on other OS / browsers?

**Reply by tom-deadline** [reply]
I am working on Linux, but I have tested on other os/browsers too and the result is the same

**Reply by tom-deadline** [reply]
I have tried loading the model with react-three-fiber and if I set devicePixelRatio to 2 and antialiasing to true , it helps a lot, moire artifacts are only slightly visible on some angles with these settings. Is there a way to set antialiasing to true in the model-viewer?

**Reply by elalish** [reply]
antialiasing is already set to true: https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/three-components/Renderer.ts#L127

**Reply by tom-deadline** [reply]
Than how can I set dpr to 2?

**Reply by elalish** [reply]
model-viewer uses the browser DPR. Doubling this will reduce performance by a factor of 4. I would highly encourage tweaking the model (reducing the normal map intensity or making that part a bit less shiny). We attempt to anti-alias based on the normal map, but there are tradeoffs.
